### PR TITLE
Added grpc_client_handed_total prometheus metric for datacatalog

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR
Slack conversation on the context

https://flyte-org.slack.com/archives/CNMKCU6FR/p1652437798343639

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

https://github.com/flyteorg/flyte/issues/2489

## Follow-up issue
_NA_

### CODE REVIEW

**Review by hamersaw — APPROVED** [review]

**Comment by sonjaer** [comment]
Thank you for fixing this!

**Review by EngHabu — APPROVED** [review]
